test/k8sT/Policies.go

```diff
@@ -808,3 +815,284 @@ var _ = Describe("K8sValidatedPolicyTestAcrossNamespaces", func() {
 	}, 300)
 
 })
+
+// Create a server pod with a listening container for each port in ports[].
+// Will also assign a pod label with key: "pod-name" and label set to the given podname for later use by the network
+// policy.
+func createServerPodAndService(k *helpers.Kubectl, namespace, podName string, ports []int) (*v1.Pod, *v1.Service) {
+	// Because we have a variable amount of ports, we'll first loop through and generate our Containers for our pod,
+	// and ServicePorts.for our Service.
+	containers := []v1.Container{}
+	servicePorts := []v1.ServicePort{}
+	for _, port := range ports {
+		// Build the containers for the server pod.
+		containers = append(containers, v1.Container{
+			Name:  fmt.Sprintf("%s-container-%d", podName, port),
+			Image: "porter:1.0",
+			Env: []v1.EnvVar{
+				{
+					Name:  fmt.Sprintf("SERVE_PORT_%d", port),
+					Value: "foo",
+				},
+			},
+			Ports: []v1.ContainerPort{
+				{
+					ContainerPort: int32(port),
+					Name:          fmt.Sprintf("serve-%d", port),
+				},
+			},
+			ReadinessProbe: &v1.Probe{
+				Handler: v1.Handler{
+					HTTPGet: &v1.HTTPGetAction{
+						Path: "/",
+						Port: intstr.IntOrString{
+							IntVal: int32(port),
+						},
+						Scheme: v1.URISchemeHTTP,
+					},
+				},
+			},
+		})
+
+		// Build the Service Ports for the service.
+		servicePorts = append(servicePorts, v1.ServicePort{
+			Name:       fmt.Sprintf("%s-%d", podName, port),
+			Port:       int32(port),
+			TargetPort: intstr.FromInt(port),
+		})
+	}
+
+	By(fmt.Sprintf("Creating a server pod %s in namespace %s", podName, namespace))
+	pod, err := k.CoreV1().Pods(namespace).Create(&v1.Pod{
+		ObjectMeta: metav1.ObjectMeta{
+			Name: podName,
+			Labels: map[string]string{
+				"pod-name": podName,
+			},
+		},
+		Spec: v1.PodSpec{
+			Containers:    containers,
+			RestartPolicy: v1.RestartPolicyNever,
+		},
+	})
+	Expect(err).NotTo(HaveOccurred())
+
+	svcName := fmt.Sprintf("svc-%s", podName)
+	By(fmt.Sprintf("Creating a service %s for pod %s in namespace %s", svcName, podName, namespace))
+	svc, err := k.CoreV1().Services(namespace).Create(&v1.Service{
```

Comment:
I'm thinking if we create a pods/services like this, you'll never know what is going to be applied/created. No log for the pods/services manifest and not sure if you can repreduce the error in the future. Could we have a log or something in here?

Reply:
At the moment, each time that something is executed is logged on `test_result/${TESTNAME}/logs` with the new way that you added, some pods, services will be created without having that info in the logs.

I would like to have something: 
```
DEBUG= "HTTPRequest" to="URL" method="POST" data=""  result=''
```
in the logs so we can examine what has been applied into the service, for example: 
```
level=debug msg="running command: kubectl apply -f  /src/test//k8sT/manifests/demo.yaml"
cmd: "kubectl apply -f  /src/test//k8sT/manifests/demo.yaml" exitCode: 0 
 service "app1-service" created
deployment "app1" created
deployment "app2" created
deployment "app3" created
```

```
level=debug msg="running command: kubectl exec -n kube-system cilium-z6hcp -- cilium policy trace --src-k8s-pod default:app2-6cc47dd5b6-9s6bv --dst-k8s-pod default:app1-54db9c4479-bfnmr --dport 80"
cmd: "kubectl exec -n kube-system cilium-z6hcp -- cilium policy trace --src-k8s-pod default:app2-6cc47dd5b6-9s6bv --dst-k8s-pod default:app1-54db9c4479-bfnmr --dport 80" exitCode: 0 
 ----------------------------------------------------------------
Tracing From: [k8s:io.kubernetes.pod.namespace=default, k8s:zgroup=testapp, k8s:id=app2] => To: [k8s:id=app1, k8s:zgroup=testapp, k8s:io.kubernetes.pod.namespace=default] Ports: [80/ANY]
* Rule {"matchLabels":{"k8s:id":"app1","k8s:io.kubernetes.pod.namespace":"default"}}: selected
    Allows from labels {"matchLabels":{"k8s:id":"app2","k8s:io.kubernetes.pod.namespace":"default"}}
      Found all required labels
        Rule restricts traffic to specific L4 destinations; deferring policy decision to L4 policy stage
1/1 rules selected
Found no allow rule
Label verdict: undecided

Resolving egress port policy for [k8s:io.kubernetes.pod.namespace=default k8s:zgroup=testapp k8s:id=app2]
0/1 rules selected
Found no allow rule
L4 egress verdict: undecided

Resolving ingress port policy for [k8s:id=app1 k8s:zgroup=testapp k8s:io.kubernetes.pod.namespace=default]
* Rule {"matchLabels":{"k8s:id":"app1","k8s:io.kubernetes.pod.namespace":"default"}}: selected
    Allows Ingress port [{80 TCP}] from endpoints [{"matchLabels":{"k8s:id":"app2","k8s:io.kubernetes.pod.namespace":"default"}}]
      Found all required labels
1/1 rules selected
Found allow rule
L4 ingress verdict: allowed

Final verdict: ALLOWED
level=info msg="Waiting for k8s api-server to be ready..." subsys=k8s
level=info msg="Connected to k8s api-server" ipAddr="https://10.96.0.1:443" subsys=k8s
level=info msg="Waiting for k8s api-server to be ready..." subsys=k8s
level=info msg="Connected to k8s api-server" ipAddr="https://10.96.0.1:443" subsys=k8s

level=debug msg="running command: kubectl exec -n kube-system cilium-z6hcp -- cilium policy trace --src-k8s-pod default:app3-596cc57c5b-7ffgp --dst-k8s-pod default:app1-54db9c4479-bfnmr"
cmd: "kubectl exec -n kube-system cilium-z6hcp -- cilium policy trace --src-k8s-pod default:app3-596cc57c5b-7ffgp --dst-k8s-pod default:app1-54db9c4479-bfnmr" exitCode: 0 
```